cmd/egnaddrs/main.go

```
@@ -111,18 +113,25 @@ func getRegCoordAndServiceMngrAddr(
 	registryCoordinatorAddrString := c.String(RegistryCoordinatorAddrFlag.Name)
 	if registryCoordinatorAddrString != "" {
 		registryCoordinatorAddr := common.HexToAddress(registryCoordinatorAddrString)
-		registryCoordinatorC, err := regcoord.NewContractRegistryCoordinator(
-			registryCoordinatorAddr,
-			client,
-		)
-		if err != nil {
-			return common.Address{}, common.Address{}, err
-		}
-		serviceManagerAddr, err := registryCoordinatorC.ServiceManager(&bind.CallOpts{})
-		if err != nil {
-			return common.Address{}, common.Address{}, err
+
+		serviceManagerAddrString := c.String(ServiceManagerAddrFlag.Name)
+		if serviceManagerAddrString == "" {
+			registryCoordinatorC, err := regcoord.NewContractRegistryCoordinator(
+				registryCoordinatorAddr,
+				client,
+			)
+			if err != nil {
+				return common.Address{}, common.Address{}, err
+			}
+
+			serviceManagerAddr, err := registryCoordinatorC.ServiceManager(&bind.CallOpts{})
+			if err != nil {
+				return registryCoordinatorAddr, common.Address{}, nil
+			}
```

Please add a comment explaining why we ignore this error.

---

Done in [b1298c8](https://github.com/Layr-Labs/eigensdk-go/pull/585/commits/b1298c82d894a6e68019bb5ca0b8f246abf7cc59)